src/raster/rasterband.rs

```diff
@@ -600,3 +609,143 @@ impl ColorInterpretation {
         _string(rv)
     }
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PaletteInterpretation {
+    Gray,
+    Rgba,
+    Cmyk,
+    Hls,
+}
+
+impl PaletteInterpretation {
+    fn from_c_int(palette_interpretation: GDALPaletteInterp::Type) -> Option<Self> {
+        match palette_interpretation {
+            GDALPaletteInterp::GPI_Gray => Some(Self::Gray),
+            GDALPaletteInterp::GPI_RGB => Some(Self::Rgba),
+            GDALPaletteInterp::GPI_CMYK => Some(Self::Cmyk),
+            GDALPaletteInterp::GPI_HLS => Some(Self::Hls),
+            _ => None,
+        }
+    }
+}
```

Comment:
```suggestion
impl PaletteInterpretation {
    fn from_c_int(palette_interpretation: GDALPaletteInterp::Type) -> Self {
        match palette_interpretation {
            GDALPaletteInterp::GPI_Gray => Self::Gray,
            GDALPaletteInterp::GPI_RGB => Self::Rgba,
            GDALPaletteInterp::GPI_CMYK => Self::Cmyk,
            GDALPaletteInterp::GPI_HLS => Self::Hls,
            _ => unreachable!("GDAL has implemented a new type of `GDALPaletteInterp`"),
        }
    }
}
```

If you unwrap it anyway, you could just panic here.